src/MobileRedirection.php

```diff
@@ -165,21 +177,54 @@ public function redirect() {
 			}
 
 			// Add a link to the footer to allow for navigation to the AMP version.
-			add_action( 'wp_footer', [ $this, 'add_mobile_version_switcher_link' ] );
+			$this->add_mobile_switcher_footer_hooks();
 		} else {
 			if ( ! $js && $this->is_redirection_disabled_via_cookie() ) {
 				$this->set_mobile_redirection_disabled_cookie( false );
 			}
 
-			add_filter( 'amp_to_amp_linking_element_excluded', [ $this, 'filter_amp_to_amp_linking_element_excluded' ], 100, 2 );
-			add_filter( 'amp_to_amp_linking_element_query_vars', [ $this, 'filter_amp_to_amp_linking_element_query_vars' ], 10, 2 );
+			$this->add_a2a_linking_hooks();
 
 			// Add a link to the footer to allow for navigation to the non-AMP version.
-			add_action( 'wp_footer', [ $this, 'add_mobile_version_switcher_link' ] );
-			add_action( 'amp_post_template_footer', [ $this, 'add_mobile_version_switcher_link' ] ); // For legacy Reader mode theme.
+			$this->add_mobile_switcher_footer_hooks();
 		}
 	}
 
+	/**
+	 * Add mobile version switcher link/styles when mobile redirection is not enabled.
+	 */
+	public function maybe_add_mobile_switcher_hooks() {
+		if ( $this->is_mobile_request() ) {
+			$this->add_mobile_switcher_head_hooks();
+			$this->add_a2a_linking_hooks();
+			$this->add_mobile_switcher_footer_hooks();
+		}
+	}
```

Comment:
Per above, it turns out that we don't need this. 

```suggestion
```

Also, the use of `is_mobile_request()` here wouldn't work reliably when JS-based redirection is being used since a caching layer could be serving a page that was originally accessed by desktop to a mobile visitor.